content/en/docs/concepts/scheduling-eviction/assign-pod-node.md

```diff
@@ -358,6 +358,88 @@ The affinity term is applied to namespaces selected by both `namespaceSelector`
 Note that an empty `namespaceSelector` ({}) matches all namespaces, while a null or empty `namespaces` list and
 null `namespaceSelector` matches the namespace of the Pod where the rule is defined.
 
+#### matchLabelKeys
+
+{{< feature-state for_k8s_version="v1.29" state="alpha" >}}
+
+`matchLabelKeys` is an optional field which specifies the keys for the labels that should match with the incoming Pod's labels, when satisfying the Pod (anti)affinity.
+
+{{< note >}}
+The `matchLabelKeys` field is a alpha-level field and disabled by default in 1.29. 
+When you want to use it, you have to enable it via the
+`MatchLabelKeysInPodAffinity` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/).
+{{< /note >}}
+
+The keys are used to lookup values from the pod labels, those key-value labels are ANDed 
+with labelSelector to select the group of existing pods which will be taken into Pod (anti)affinity calculation. 
+
+One common usecase would be to use `matchLabelKeys` with `pod-template-hash` in Deployment 
+so that the rolling upgrade won't break affinity.
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: application-server
+...
+  affinity:
+    podAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+      - labelSelector:
+          matchExpressions:
+          - key: app
+            operator: In
+            values:
+            - database
+        topologyKey: topology.kubernetes.io/zone
+        # Only Pods of the same version are taken into consideration of PodAffinity calculation.
+        matchLabelKeys: 
+        - pod-template-hash
+```
+
+#### mismatchLabelKeys
+
+{{< feature-state for_k8s_version="v1.29" state="alpha" >}}
+
+mismatchLabelKeys is an optional field 
+which specifies the keys for the labels that should NOT match with the incoming Pod's labels, 
+when satisfying the Pod (anti)affinity.
+
+{{< note >}}
+The `mismatchLabelKeys` field is a alpha-level field and disabled by default in 1.29. 
+When you want to use it, you have to enable it via the
+`MatchLabelKeysInPodAffinity` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/).
+{{< /note >}}
+
+One usecase is to ensure Pods go to the domain where only Pods from the same group are scheduled in.
```

Comment:
```suggestion
One example use case is to ensure Pods go to the domain where only Pods from the same group are scheduled in.
In other words, you want to avoid running Pods from two different tenants on the same node at the same time.
```